backend/src/main/java/com/team4/backend/service/MonitorService.java

```diff
@@ -26,10 +24,10 @@ public MonitorService(MonitorRepository monitorRepository, UserService userServi
         this.pbkdf2Encoder = pbkdf2Encoder;
     }
 
-    public Mono<Monitor> findMonitorByEmail(String email) {
-        return monitorRepository.findByEmail(email)
-                .filter(Objects::nonNull)
-                .switchIfEmpty(Mono.error(new ResponseStatusException(HttpStatus.NOT_FOUND, "Can't find monitor with this email.")));
+    public Mono<Boolean> existsByEmailAndIsEnabledTrue(String email) {
+        return monitorRepository.existsByEmailAndIsEnabledTrue(email)
+                .filter(exist -> exist)
+                .switchIfEmpty(Mono.error(DoNotExistException::new));
```

Comment:
Why filter return an exception when False is the intended behavior ?

Reply:
because it will go through without throwing an exception

Reply:
https://github.com/0xD34DC0DE/ProjetIntegreTeam4/pull/18/commits/e4d3c34bf0758d1781144b35a26d663b16a4fc9c